modules/services/mapillary.js

```diff
@@ -636,195 +576,138 @@ export default {
     },
 
 
+    // Return the currently displayed image
     getActiveImage: function() {
         return _mlyActiveImage;
     },
 
 
-    getSelectedImageKey: function() {
-        return _mlySelectedImageKey;
-    },
-
-
-    getSequenceKeyForImageKey: function(imageKey) {
-        return _mlyCache.sequences.forImageKey[imageKey];
+    // Return a list of detection objects for the given id
+    getDetections: function(id) {
+        return loadData(`${apiUrl}/${id}/detections?access_token=${accessToken}&fields=id,value,image`);
     },
 
 
-    setActiveImage: function(node) {
-        if (node) {
+    // Set the currently visible image
+    setActiveImage: function(image) {
+        if (image) {
             _mlyActiveImage = {
-                ca: node.originalCA,
-                key: node.key,
-                loc: [node.originalLatLon.lon, node.originalLatLon.lat],
-                pano: node.pano
+                ca: image.originalCompassAngle,
+                id: image.id,
+                loc: [image.originalLngLat.lng, image.originalLngLat.lat],
+                is_pano: image.cameraType === 'spherical',
+                sequence_id: image.sequenceId
             };
         } else {
             _mlyActiveImage = null;
         }
-
     },
 
 
-    // Updates the currently highlighted sequence and selected bubble.
-    // Reset is only necessary when interacting with the viewport because
-    // this implicitly changes the currently selected bubble/sequence
-    setStyles: function(context, hovered, reset) {
-        if (reset) {  // reset all layers
-            context.container().selectAll('.viewfield-group')
-                .classed('highlighted', false)
-                .classed('hovered', false);
-
-            context.container().selectAll('.sequence')
-                .classed('highlighted', false)
-                .classed('currentView', false);
-        }
-
-        var hoveredImageKey = hovered && hovered.key;
-        var hoveredSequenceKey = hoveredImageKey && this.getSequenceKeyForImageKey(hoveredImageKey);
-        var hoveredLineString = hoveredSequenceKey && _mlyCache.sequences.lineString[hoveredSequenceKey];
-        var hoveredImageKeys = (hoveredLineString && hoveredLineString.properties.coordinateProperties.image_keys) || [];
-
-        var selectedImageKey = _mlySelectedImageKey;
-        var selectedSequenceKey = selectedImageKey && this.getSequenceKeyForImageKey(selectedImageKey);
-        var selectedLineString = selectedSequenceKey && _mlyCache.sequences.lineString[selectedSequenceKey];
-        var selectedImageKeys = (selectedLineString && selectedLineString.properties.coordinateProperties.image_keys) || [];
-
-        // highlight sibling viewfields on either the selected or the hovered sequences
-        var highlightedImageKeys = utilArrayUnion(hoveredImageKeys, selectedImageKeys);
+    // Update the currently highlighted sequence and selected bubble.
+    setStyles: function(context, hovered) {
```

Comment:
I've noticed a very small inconsistency here [not sure if it is a inconsistency or a feature 😁]
Repro: Toggle Mapillary imagery on, select one viewport (color to yellow), toggle imagery off.
The selected item stays selected and visible.
I saw that OpenStreetCam handles this by clearing all.

Actually I think it would be clean from ui perspective, to clear all and also clear the 'view photo' box.

Reply:
That was intentional, the current position of the image remains visible without the Mapillary imagery layer (it's a "separate" layer in mapillary_position.js). This is because you can turn on only traffic signs (for example) and then you can click on a traffic sign and see the position of the image looking at the sign even without turning on the imagery layer. This goal of this feature is to help the user figure out the actual position of the traffic sign based on the  detected sign, the image in the viewer and the image position marker on the map.